pkg/database/database.go

```diff
@@ -32,18 +35,33 @@ type Database struct {
 	resourceTableName string
 }
 
-func NewDatabase() (*Database, error) {
+func GetDatabaseConnection() (*Database, error) {
 	dataSource, err := ConnectionStr()
 	if err != nil {
 		return nil, err
 	}
-	db, err := otelsql.Open("postgres", dataSource)
-	if err != nil {
-		return nil, err
+	var db *sql.DB
+	configs := []retry.Option{
+		retry.Attempts(10),
+		retry.OnRetry(func(n uint, err error) {
+			log.Printf("Retry request %d, get error: %v", n+1, err)
+		}),
+		retry.Delay(time.Second),
```

Comment:
The delay should be longer than a single second. I would suggest at least 5 seconds between each retry. I don't know if it is supported in the library you are using, but I think an exponential back-off would be appropriate here.

Reply:
I'm sorry, I should add to the PR notes that, by default, the library uses exponential back-off.
These are some traces from the integration-test
`2024/08/12 15:17:46 2024/08/12 **13:17:03** Retry request 1, get error: dial tcp 10.96.117.213:5432: connect: connection timed out
2024/08/12 **13:17:04** Retry request 2, get error: dial tcp 10.96.117.213:5432: connect: connection refused
2024/08/12 **13:17:06** Retry request 3, get error: dial tcp 10.96.117.213:5432: connect: connection refused
2024/08/12 **13:17:10** Retry request 4, get error: dial tcp 10.96.117.213:5432: connect: connection refused
2024/08/12 **13:17:18** Successfully connected to the database
`

Reply:
You don't need a comment on the PR. If Delay does exponential back-off by default that is fine.